libraries/AP_InertialSensor/AP_InertialSensor_Invensense.cpp

```diff
@@ -714,7 +714,16 @@ void AP_InertialSensor_Invensense::_read_fifo()
             }
         } else {
             // this ensures we keep things nicely setup for DMA
-            uint8_t reg = MPUREG_FIFO_R_W | 0x80;
+            // NOTE: Fast sampling doesn't work with 0x80 mask
```

Comment:
you're right that this isn't right for I2C, but the fix is also not quite right. The fix should check if we're on I2C, not if we're doing fast sampling.
Also, right now set_chip_select() always returns false on i2c (as it makes no sense for i2c), so the bug can't happen.
What device setup did you have that triggered this?

Reply:
yes, i thought to do that, checking the i2c bus type, but it's fixed and known when start() call _set_filter_register(), i don't know why we are using a type fixed bitmask parameter, i thought that was a special case on some boards.

I still using the APM 2.x board :(

Reply:
Ahh Tridge, i detected that you are reading the fifo register with the correct addr only one time on spi and the rest array doesn't send the fifo addr register, it's send 0x00 addr. But i'm not sure if your board are sending the correct addr on each read.

Reply:
i forgot to tell you that i test i2c on a generic board with MPU6050.